kart/spatial_filters.py

```diff
@@ -134,36 +155,67 @@ def __init__(self, ref_or_oid):
         super().__init__()
         self.ref_or_oid = ref_or_oid
 
-    def write_config(self, repo):
+    def _resolve_object_contents(self, obj):
+        contents = obj.data.decode("utf-8")
+        parts = self.split_file(contents)
+        return ResolvedSpatialFilterSpec(*parts)
+
+    @functools.lru_cache(maxsize=1)
+    def _resolve_target(self, repo):
+        # Returns a tuple of strings (reference, object_id, ResolvedSpatialFilterSpec).
+        # Returned reference will be None if ref_or_oid is an object-id.
         obj = None
+        oid = self.ref_or_oid
         try:
-            obj = repo[self.ref_or_oid]
+            obj = repo[oid]
         except (KeyError, ValueError):
             pass
 
         if obj is not None:
+            return None, oid, self._resolve_object_contents(obj)
+
+        ref = self.ref_or_oid
+        if not ref.startswith("refs/"):
+            ref = f"refs/filters/{ref}"
+
+        if ref in repo.references:
+            oid = str(repo.references[ref].resolve().target)
+            try:
+                obj = repo[oid]
+            except (KeyError, ValueError):
+                pass
+
+        if obj is not None:
+            return ref, oid, self._resolve_object_contents(obj)
+
+        ref_desc = " or ".join(set([oid, ref]))
+        raise NotFound(
+            f"No spatial filter object was found in the repository at {ref_desc}",
+            exit_code=NO_SPATIAL_FILTER,
+        )
+
+    def resolve(self, repo):
+        # Returns the ResolvedSpatialFilterSpec that this ReferenceSpatialFilterSpec points to.
```

Comment:
This comment needs clarifying I think (and turning into a docstring)

Reply:
Expanded the docstring and moved it to a superclass.